flux/src/main/java/software/amazon/aws/clients/swf/flux/poller/WorkflowState.java

```diff
@@ -293,17 +296,28 @@ public static WorkflowState build(PollForDecisionTaskResponse task) {
             } else if (MARKER_EVENTS.contains(event.eventType())) {
                 // There may be markers we don't care about in the history, such as the "unknown result code" marker that Flux adds,
                 // or markers added by other tools.
-                String prefix = TaskNaming.PARTITION_METADATA_MARKER_NAME_PREFIX;
                 String markerName = event.markerRecordedEventAttributes().markerName();
-                if (markerName != null && markerName.startsWith(prefix)) {
-                    // TODO -- handle data split across multiple markers.
-
-                    // We need to extract the activity name from the marker name, which is of the form "{prefix}.{stepName}".
-                    String stepName = markerName.substring(prefix.length() + 1);
-
-                    // It's possible the marker was added more than once for some reason.
-                    // If it was, we want to use the most recent one, so only insert if we don't already have a record for it.
-                    ws.rawPartitionMetadata.putIfAbsent(stepName, getMarkerData(event));
+                if (TaskNaming.isPartitionMetadataMarker(markerName)) {
+                    String stepName = TaskNaming.extractPartitionMetadataMarkerStepName(markerName);
+                    Long subsetId = TaskNaming.extractPartitionMetadataMarkerSubsetId(markerName);
+
+                    partitionMarkerSubsetsFound.putIfAbsent(stepName, new HashSet<>());
+                    partitionMarkerSubsetsFound.get(stepName).add(subsetId);
```

Comment:
Seeing this makes me think that it would certainly make things look simplier if you just used a multimap. though, functionally they are the same here.

Reply:
We don't have a multimap implementation available without pulling in a new dependency.